include/rdma/rdma_cma.h

```diff
@@ -197,6 +197,29 @@ struct rdma_addrinfo {
 	struct rdma_addrinfo	*ai_next;
 };
 
+/* Multicast join compatibility mask attributes */
+enum rdma_cm_join_mc_attr_mask {
+	RDMA_CM_JOIN_MC_ATTR_ADDRESS	= 1 << 0,
+	RDMA_CM_JOIN_MC_ATTR_JOIN_FLAGS	= 1 << 1,
+	RDMA_CM_JOIN_MC_ATTR_RESERVED	= 1 << 2,
+};
+
+/* Multicast join flags */
+enum rdma_cm_mc_join_flags {
+        RDMA_MC_JOIN_FLAG_FULLMEMBER,
+        RDMA_MC_JOIN_FLAG_SENDONLY_FULLMEMBER,
+        RDMA_MC_JOIN_FLAG_RESERVED,
+};
```

Comment:
After some thinking and reading your comments (more than once), 
the main problem I see is that SENDONLY != SENDONLY_FULLMEMBER.
I implemented kernel code support for SENDONLY_FULLMEMBER,
using a flag called SENDONLY can be misleading since it might
also be a future valid option (already in IB spec.).

Regarding the bit values, currently we provide in the command struct a user value for the 
join state, the kernel translates it to the kernel supported value in ucma based on enums/defines in  
rdma_user_cm.h this means that both should be aligned. To use bit values we will need to:
Modify rdma_user_cm.h in the kernel or add a redundant enum that translates the value in librdmacm.

The solution below is a combination of your suggestion with minor changes:
Prototype: rdma_join_multicast_ex(.., join_flags)
Flags: FULLMEMBER or SENDONLY_FULLMEMBER which are bit value
The flags above will be translated based on the values in rdma_user_cm.h (bit value)
which means we don't need to change the kernel and wait another cycle.